volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -1428,3 +1429,110 @@ def get_type(self):
 
         # kernel < 3.18.140
         raise AttributeError("Unable to find the BPF type")
+
+
+class cred(objects.StructType):
+    # struct cred was added in kernels 2.6.29
+    def _get_cred_int_value(self, member: str) -> int:
+        """Helper to obtain the right cred member value for the current kernel.
+
+        Args:
+            member (str): The requested cred member name to obtain its value
+
+        Raises:
+            AttributeError: When the requested cred member doesn't exist
+            AttributeError: When the cred implementation is not supported.
+
+        Returns:
+            int: The cred member value
+        """
+        if not self.has_member(member):
+            raise AttributeError(f"struct cred doesn't have a '{member}' member")
+
+        cred_val = self.member(member)
+        if hasattr(cred_val, "val"):
+            # From kernels 3.5.7 on it is a 'kuid_t' type
+            value = cred_val.val
+        elif isinstance(cred_val, objects.Integer):
+            # From at least 2.6.30 and until 3.5.7 it was a 'uid_t' type which was an 'unsigned int'
+            value = cred_val
+        else:
+            raise AttributeError("Kernel struct cred is not supported")
+
+        return int(value)
```

Comment:
It *should* be that value *is* an int already, so unless you're strictly doing this cast for type checking reasons that you've run into, this cast shouldn't be here (and I'm undecided about what return type I'd prefer, leaving it as `int` will probably throw off mypy, but should theoretically be accurate.).

Reply:
I'll let this slide (but only because you've been so good on getting the rest of it all sorted!)  5;P